dll/win32/msafd/misc/dllmain.c

```diff
@@ -3349,47 +3470,53 @@ GetSocketInformation(PSOCKET_INFORMATION Socket,
                                    APCContext,
                                    IOSB,
                                    IOCTL_AFD_GET_INFO,
-                                   &InfoData,
-                                   sizeof(InfoData),
-                                   &InfoData,
-                                   sizeof(InfoData));
-
-    /* Wait for return */
-    if (Status == STATUS_PENDING && Overlapped == NULL)
+                                   InfoData,
+                                   sizeof(AFD_INFO),
+                                   InfoData,
+                                   sizeof(AFD_INFO));
+    if (Status == STATUS_PENDING)
     {
-        WaitForSingleObject(SockEvent, INFINITE);
+        if (!Overlapped)
+        {
+            /* Wait for completion of not overlapped */
+            WaitForSingleObject(SockEvent, INFINITE);
+        }
+
         Status = IOSB->Status;
     }
 
-    NtClose( SockEvent );
-
-    TRACE("Status %x Information %d\n", Status, IOSB->Information);
+    TRACE("Status %x\n", Status);
 
-    if (Status == STATUS_PENDING)
+    if (Status == STATUS_SUCCESS)
     {
-        TRACE("Leaving (Pending)\n");
-        return WSA_IO_PENDING;
+        /* Return Information */
+        if (Ulong != NULL)
+        {
+            *Ulong = InfoData->Information.Ulong;
+        }
+        if (LargeInteger != NULL)
```

Comment:
Can all these "Ulong" / "LargeInteger" / ... pointers be not NULL at the same time?

Reply:
Not sure, the code is the same as original.